Fix an inconsistent datum issue

LAGO-Support opened the pull request:
Data from DFE files have inconsistent datums and this fix will shift NAVD88 datums to NGVD29. Subsequent scripts will read the converted data.

LAGO-Support committed:
Fix an inconsistent datum issue

Data from DFE files have inconsistent datums and this fix will shift NAVD88 datums to NGVD29. Subsequent scripts will read the converted data.

LAGO-Support commented:
Conversion to NGVD29 only occurs if the station datum is NAVD88 and the conversion value is not 0.

kirenbahm commented:
Hi Kham,
Thanks for your work on this important edit. The timeseries values look good, but I would like something slightly different in the header of the ascii output. I think the vertical datum and ground surface elevation should all be NGVD29. I have put an example below. Let me know if this change can be implemented and still work correctly with the downstream code. 

===============================
INPUT FILE:
===============================
#station: LS
#agency: NPS_ENP
#ground surface elevation ft: -3.87
#latitude: 25.23516
#longitude: -80.45680
#vertical datum: NAVD88
#conversion: -1.539

#stn|type|date time|value(feet)|validation_date
LS|stage|2002-07-21 13:10|-0.6890|2019-08-21
LS|stage|2002-07-21 13:20|-0.6890|2019-08-21
LS|stage|2002-07-21 13:30|-0.6890|2019-08-21
LS|stage|2002-07-21 13:40|-0.6890|2019-08-21
LS|stage|2002-07-21 13:50|-0.6790|2019-08-21
LS|stage|2002-07-21 14:00|-0.6790|2019-08-21
LS|stage|2002-07-21 14:10|-0.6790|2019-08-21
LS|stage|2002-07-21 14:20|-0.6790|2019-08-21

===============================
OUTPUT FILE (After Lago Edits):
===============================
#station: LS
#agency: NPS_ENP
#ground surface elevation ft: -3.87
#latitude: 25.23516
#longitude: -80.45680
#vertical datum: NAVD88 (Will Convert values to NGVD29 Datum)
#conversion: -1.539

#stn|type|date time|value(feet)|validation_date|Flag
LS|stage|2002-07-21 13:10|0.8500|2019-08-21|
LS|stage|2002-07-21 13:20|0.8500|2019-08-21|
LS|stage|2002-07-21 13:30|0.8500|2019-08-21|
LS|stage|2002-07-21 13:40|0.8500|2019-08-21|
LS|stage|2002-07-21 13:50|0.8600|2019-08-21|
LS|stage|2002-07-21 14:00|0.8600|2019-08-21|
LS|stage|2002-07-21 14:10|0.8600|2019-08-21|
LS|stage|2002-07-21 14:20|0.8600|2019-08-21|

===============================
DESIRED OUTPUT FILE:
===============================
#station: LS
#agency: NPS_ENP
#ground surface elevation ft: -2.33
#latitude: 25.23516
#longitude: -80.45680
#vertical datum: NGVD29
#conversion: -1.539

#stn|type|date time|value(feet)|validation_date|Flag
LS|stage|2002-07-21 13:10|0.8500|2019-08-21|
LS|stage|2002-07-21 13:20|0.8500|2019-08-21|
LS|stage|2002-07-21 13:30|0.8500|2019-08-21|
LS|stage|2002-07-21 13:40|0.8500|2019-08-21|
LS|stage|2002-07-21 13:50|0.8600|2019-08-21|
LS|stage|2002-07-21 14:00|0.8600|2019-08-21|
LS|stage|2002-07-21 14:10|0.8600|2019-08-21|
LS|stage|2002-07-21 14:20|0.8600|2019-08-21|

LAGO-Support committed:
adjust header info in flagged *.dat files 

In the case that *.dat file has its values converted from the NAVD88 datum to the NGVD29, the header will now have the datum as NGVD29 instead of NAVD88 with a note that the values were converted. The ground surface elevation in the header will also be converted from NAVD88 to NGVD29.

LAGO-Support committed:
fix latitude header line in output

The error was misplacing the latitude line in output *.dat file header.

LAGO-Support committed:
fix an issue with null values being replaced

This change involves the case where flagged *.dat files are written if a value was null in the raw *.dat file and triggers the "not a number" flag. When the value is written to the output flagged *.dat file a null value will be written. When written to the dfs0 file, the value will be replaced with the no data value of -1e-35.
